### PR TITLE
Update prysm.sh to include slasher and sig verify

### PR DESCRIPTION
**Pull request by prestonvanloon**
Key changes:
- Only download the binary for the process you are attempting to run.
- Add slasher option
- Verify sha256 of binary
- Verify signature of binary

### CODE REVIEW

**Comment by Yoldark34** [comment]
Hello, is it normal that gpg is telling the RSA key "0AE0051D647BA3C1A917AF4072E33E4DF1A5036E" is not public ?

I had to do "gpg --recv-keys 0AE0051D647BA3C1A917AF4072E33E4DF1A5036E". If it's normal it may be usefull to ask the key before trying to check the signature or at least display the command the user has to run in order to get the missing key.

Best regards.

**Comment by Yoldark34** [comment]
Not that is my business but IMO you should have made 2 separate pullRequest for the commit c2bfe24, you are adding two separate functionality in one PR.

**Comment by prestonvanloon** [comment]
@Yoldark34 Thanks for the feedback. In general, I think that these features are relatively small in scope and shouldn't be much of a burden on reviewers to review in a single PR. With that said, I probably still should have split this into multiple PRs as I have listed 4 key changes. If a reviewer feels strongly about this, I can split it up.

**Comment by farazdagi** [comment]
I am getting "sha256sum is not available. Not verifying integrity of downloaded binary." error, am I expected to download sth manually?

**Comment by Yoldark34** [comment]
I don't think it is a good idea for a script to install third party software so, yes, you have to install it yourself

**Comment by prestonvanloon** [comment]
Sounds like mac has shasum installed by default, we can use `shasum -a 256` do achieve the same goal. 

With regards to the public key, I think i'll just make it part of the prysm.sh file so it doesn't need to retrieve from a key server, which could be offline or compromised. 

@Yoldark34, I'll give it a try. If Y/N input is too difficult to do, maybe we can refuse to run an unverified binary unless there is SKIP_PRYSM_VERIFY=1 or something along those lines that requires user input to skip a failed verification.

File: prysm.sh [hunk 1]
```diff
@@ -108,18 +108,32 @@ function get_prysm_version() {
...
+  if [[ $skip == 1 ]]; then
+    return 0
+  fi
+  color "31" "Failed to verify Prysm binary. Please erase downloads in the \
+dist directory and run this script again. Alternatively, you can use a \
+A prior version by specifying environment variable USE_PRYSM_VERSION \
+with the specific version, as desired. Example: USE_PRYSM_VERSION=v1.0.0-alpha.5 \
+If you must wish to continue running an unverified binary, specific the \
+environment variable PRYSM_ALLOW_UNVERIFIED_BINARIES=1"
+  exit 1
```
Comment (Yoldark34) [inline prysm.sh:134]: Very nice thanks :)

**Pull request merged**